core/arch/arm/kernel/thread_spmc.c

```diff
@@ -1465,15 +1467,44 @@ static int set_pages(struct ffa_address_range *regions,
 	return 0;
 }
 
+static struct ffa_mem_transaction *spmc_frag_retrieve(uint64_t cookie,
+			 uint32_t next_fragment_offset, uint32_t *fragment_length)
+{
+	struct thread_smc_args args = {
+		.a0 = FFA_MEM_FRAG_RX,
+		.a1 = (uint32_t)(cookie),       /* handle high addr */
```

Comment:
I'd prefer assigning `a1` and `a2` with `reg_pair_from_64(cookie, &args.a1, &args.a2)`

Reply:
Done